### PR TITLE
Refine the driver note structure

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Follow up https://github.com/apache/nuttx/pull/7837:

- drivers/node: Rename note_register to note_initialize 
- drivers/node: Rename sched_note.c to to note_driver.c
- drivers/segger: Rename nuttx/note/note_sysview.h to nuttx/segger/sysview.h

## Impact
Code refactor only

## Testing
Pass CI

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: drivers/note/note_driver.c [hunk 1]
```diff
@@ -22,57 +22,1288 @@
...
+  if (!note_isenabled_switch())
+    {
+      return;
+    }
+
+  /* Format the note */
+
+  note_common(tcb, &note.nsu_cmn, sizeof(struct note_suspend_s),
+              NOTE_SUSPEND);
+  note.nsu_state           = tcb->task_state;
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:521]:
```suggestion
  note.nsu_state = tcb->task_state;
```

File: drivers/note/note_driver.c [hunk 2]
```diff
@@ -22,57 +22,1288 @@
...
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
+void sched_note_syscall_enter(int nr, int argc, ...)
+{
+  struct note_syscall_enter_s note;
+  FAR struct tcb_s *tcb = this_task();
+  unsigned int length;
+  uintptr_t arg;
+  uint8_t *args;
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:746]:
```suggestion
  FAR uint8_t *args;
```

File: drivers/note/note_driver.c [hunk 3]
```diff
@@ -22,57 +22,1288 @@
...
+  length = SIZEOF_NOTE_BINARY(len);
+  if (length > sizeof(data))
+    {
+      length = sizeof(data);
+    }
+
+  /* Format the note */
+
+  note_common(tcb, &note->nbi_cmn, length,
+              NOTE_DUMP_BINARY);
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:897]:
```suggestion
  note_common(tcb, &note->nbi_cmn, length, NOTE_DUMP_BINARY);
```

File: drivers/note/note_driver.c [hunk 4]
```diff
@@ -22,57 +22,1288 @@
...
+ *          If 0, the filter mode is not updated.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void sched_note_filter_mode(struct note_filter_mode_s *oldm,
+                            struct note_filter_mode_s *newm)
+{
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:1199]:
```suggestion
void sched_note_filter_mode(FAR struct note_filter_mode_s *oldm,
                            FAR struct note_filter_mode_s *newm)
{
```

File: drivers/note/note_driver.c [hunk 5]
```diff
@@ -22,57 +22,1288 @@
...
+ *          If 0, the setting is not updated.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
+void sched_note_filter_syscall(struct note_filter_syscall_s *oldf,
+                               struct note_filter_syscall_s *newf)
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:1239]:
```suggestion
void sched_note_filter_syscall(FAR struct note_filter_syscall_s *oldf,
                               FAR struct note_filter_syscall_s *newf)
```

File: drivers/note/note_driver.c [hunk 6]
```diff
@@ -22,57 +22,1288 @@
...
+ *          If 0, the setting is not updated.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_IRQHANDLER
+void sched_note_filter_irq(struct note_filter_irq_s *oldf,
+                           struct note_filter_irq_s *newf)
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:1285]:
```suggestion
void sched_note_filter_irq(FAR struct note_filter_irq_s *oldf,
                           FAR struct note_filter_irq_s *newf)
```

Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:1285]: Done.

File: drivers/note/note_driver.c (hunk 5 above)
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:1239]: Done.

File: drivers/note/note_driver.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:746]: Done.

File: drivers/note/note_driver.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:521]: Done

File: drivers/note/note_driver.c [hunk 7]
```diff
@@ -22,57 +22,1288 @@
...
+   */
+
+  if (up_interrupt_context())
+    {
+#ifdef CONFIG_SCHED_INSTRUMENTATION_IRQHANDLER
+#ifdef CONFIG_SMP
+      int cpu = this_cpu();
+#else
+      int cpu = 0;
+#endif
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:275]: Done.

File: drivers/note/note_driver.c [hunk 8]
```diff
@@ -22,57 +22,1288 @@
...
+   */
+
+  if (!(g_note_filter.mode.flag & NOTE_FILTER_MODE_FLAG_IRQ) ||
+      NOTE_FILTER_IRQMASK_ISSET(irq, &g_note_filter.irq_mask))
+    {
+#ifdef CONFIG_SMP
+      int cpu = this_cpu();
+#else
+      int cpu = 0;
+#endif
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:337]: Done.

File: drivers/note/note_driver.c [hunk 9]
```diff
@@ -22,57 +22,1288 @@
...
+#ifdef CONFIG_SCHED_INSTRUMENTATION_FILTER
+  if (!(g_note_filter.mode.flag & NOTE_FILTER_MODE_FLAG_ENABLE))
+    {
+      return false;
+    }
+
+#ifdef CONFIG_SMP
+  /* Ignore notes that are not in the set of monitored CPUs */
+
+  if ((g_note_filter.mode.cpuset & (1 << this_cpu())) == 0)
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply drivers/note/note_driver.c:194]: Done.

**Comment by xiaoxiang781216** [comment]
could @acassis @pkarashchenko could you merge this PR if no other problem. https://github.com/apache/nuttx/pull/7841 depends on this.

File: drivers/note/note_driver.c [hunk 10]
```diff
@@ -22,57 +22,1276 @@
...
-      return ret;
+      return;
     }
+
+  /* Format the note */
+
+  note_common(tcb, &note.nsp_cmn, sizeof(struct note_spinlock_s), type);
+
+  sched_note_flatten(note.nsp_spinlock, &spinlock, sizeof(spinlock));
+  note.nsp_value = *(uint8_t *)spinlock;
```
Comment (pkarashchenko) [inline drivers/note/note_driver.c:415]:
```suggestion
  note.nsp_value = *(FAR uint8_t *)spinlock;
```

**Review by pkarashchenko — COMMENTED** [review]
One last comment

**Comment by xiaoxiang781216** [comment]
> One last comment

@pkarashchenko do you have more comment^_^?

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**